src/Workspaces/CSharp/Portable/Formatting/Rules/TokenBasedFormattingRule.cs

```diff
@@ -208,7 +208,7 @@ public override AdjustSpacesOperation GetAdjustSpacesOperation(SyntaxToken previ
                 }
 
                 // * ;
-                return CreateAdjustSpacesOperation(0, AdjustSpacesOption.ForceSpacesIfOnSingleLine);
+                return CreateAdjustSpacesOperation(0, AdjustSpacesOption.ForceSpaces);
```

Comment:
❗️ This change will cause the reformatting to impact _all_ formatting operations and not just the result of refactorings. I would expect the spaces to only be eliminated when the semicolon is preceded by elastic trivia (as part of code fixes/refactorings).

If we make the change here it will need to go to a design review before merge.

Reply:
@CyrusNajmabadi @sharwell The problem here is that the newline in question is *not* elastic trivia, because it's the same newline that was after the expression in the original user code.

And since, in my opinion, starting a line with a semicolon is not a reasonable code style, I thought that reformatting it everywhere would be okay.

If you disagree, I can make a more limited change.

Reply:
> @CyrusNajmabadi @sharwell The problem here is that the newline in question is not elastic trivia, because it's the same newline that was after the expression in the original user code.

An elastic trivia can always be added by the feature to say "i want to more aggressively fix up this token".

Reply:
@CyrusNajmabadi I tried adding an elastic marker and it didn't do anything. As far as I can tell, it's because `ElasticTriviaFormattingRule` is applied after `SpacingFormattingRule`, so in this case, elastic trivia doesn't get a chance to make a difference. (And changing the order would break quite a few tests.)

But while doing this, I also discovered some more cases that need to be fixed not related to semicolons (now in tests). So I changed the implementation: the coalesce refactoring now copies trailing trivia from the original node to the new node. I think that should handle all cases, and doesn't require any subsequent formatting.

Reply:
>  so in this case, elastic trivia doesn't get a chance to make a difference.

Interesting, and def seems like a bug to me.

> I think that should handle all cases, and doesn't require any subsequent formatting.

Well, that seems easy enough for me :)   I'm fine with that approach.